sdks/python/setup.py

```diff
@@ -110,7 +110,7 @@ def get_version():
   # `pipenv` package managers.
   pass
 
-REQUIRED_CYTHON_VERSION = '0.28.1'
+REQUIRED_CYTHON_VERSION = '3.0.0'
```

Comment:
This lower bound needs to be consistent with what set in base_image_requirements_manual.txt (change either)

Reply:
Going with 3.0.0

Reply:
Is there anything in beam or released Beam artifact that would make beam incompatible with Cython 0.29.0?

Reply:
If not, I suggest reverting this restriction.

Reply:
note: cython is only necessary when building beam from sources. Given that we release wheels, as far as our containers are concerned, there shouldn't be any interaction between Beam and installed version of Cython in our containers.

Reply:
This logic just warns if the cython version mismatches but doesn't actually change execution, it was already outdated